### ISSUE TITLE
Accessibility improvement: Add a role, label, and voiceover to the "close" X button

### ISSUE THREAD

**Issue by arielgreen** [Expensify/Expensify.cash#3102]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Expected Result:
The "close" X should have a role, label, and voiceover 

## Actual Result:
There is no role, label, or voiceover to the "close" X button

## Action Performed:
1. Turn on Screen Reader or Voiceover
2. Launch the Expensify.cash app or website
3. Swipe or click into any menu that has an X "close" button. For example:
- `( + )` > `New Chat` > `X`
- `( + )` >  `Request Money` > `X`
- `( + )` >  `New Group` > `X`
- tap `avatar` icon > `X`
- tap `search` 🔍  icon > `X`

4. Notice that the X "Close" button doesn't have a role, label, or voiceover component.

## Platform:
- Windows/JAWS/Chrome
- MACOs/Voiceover/Safari
- Android/Talkback
- iOS/Voiceover
- MacWebapp/Voiceover

**Notes/Photos/Videos:** 

Video - https://user-images.githubusercontent.com/37308300/119419108-7dcc8000-bcc7-11eb-9036-bb53c6e951b7.mp4

<img width=300 src=https://user-images.githubusercontent.com/13206896/120515529-63b43100-c39c-11eb-8eef-ad067d3f6a85.png>

**Upwork job link** - https://www.upwork.com/jobs/~013b962eb299963c76

[View all open jobs on Upwork](https://www.upwork.com/ab/jobs/search/?q=Expensify%20React%20Native&sort=recency&user_location_match=2)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @michaelhaxhiu (`External`), see https://stackoverflow.com/c/expensify/questions/8582 for more details.

**Comment by michaelhaxhiu** [comment]
Ugh I let this wait a few days because I wasn't clear on what to do when I initially looked at it.

I'm a little confused by what this GH is actually requesting an improvement for -- is there an associated discussion or chat? Can I please have a link? Why does "close" need a role + label?

**Comment by michaelhaxhiu** [comment]
Spoke to Ariel 1:1 and clear on this one now. Posting it shortly.

**Comment by michaelhaxhiu** [comment]
I have more questions about this:

1. What does Role & Label refer to specifically? Is that just the component that voiceover _reads_ for the user? If so, which is which in this example of fixing the "X" button voiceover?

2. I did some testing on my iPhone and Android, and noticed that the "X" doesn't have a voiceover component on **any** pages whatsoever. Should I just expand the scope of this GH to be for all "X"s on any page? 

For example, I tried:
- Green FAB + > all the options
- Settings > all the options

**Comment by parasharrajat** [comment]
@michaelhaxhiu There are couple of good reasons we want things to be accessible. Here the brief information how web will tackle this https://necolas.github.io/react-native-web/docs/accessibility/.

And https://reactnative.dev/docs/0.63/accessibility for native devices.

**Comment by michaelhaxhiu** [comment]
@parasharrajat curious what you think about scope (question #2 from above)?

**Comment by parasharrajat** [comment]
And I will say, Yes. We should combine similar behavioural components into one issue as it just require using the same code everywhere. but I am not sure how are we tracking these internally  so please take it as a opinion only.

**Comment by michaelhaxhiu** [comment]
Yep I was curious for your opinion, thanks! Appreciate the 2 cents.

@arielgreen was there a reason this was made for just the chat window? Meaning, do we need to do the improvements piece-wise like this ideally?

**Comment by arielgreen** [comment]
I would agree, let's apply this change to all 'X' buttons.

**Comment by michaelhaxhiu** [comment]
Sweet. Let's get an eng opinion before I post the job to reduce the chance of commotion of an unclear scope down the road. 

cc @stitesExpensify Can you weigh in by chance?

To make this simple, here's what I'm proposing we do with this job specifically:

## Expected Result:

>In new chat window, "Close" should have a role and a label

Update this to... Add a role, label, and voiceover to the "Close" X button

Does that make sense or does adding a role + label allow voiceover to happen automatically?

**Comment by michaelhaxhiu** [comment]
As an update, we are awaiting input on adjusting the scope of this job.

**Comment by michaelhaxhiu** [comment]
Upwork job posted here for $250 - https://www.upwork.com/jobs/~013b962eb299963c76

**Comment by dklymenk** [comment]
### Proposal

Should be as simple as adding
```
accessibilityRole="button"
accessibilityLabel="close"
```
here

https://github.com/Expensify/Expensify.cash/blob/c85ded0f70fd744ac67e53816c311b060ac68af9/src/components/HeaderWithCloseButton.js#L76-L79

and since IOU modal doesn't use the `HeaderWithCloseButton` component also add that here
https://github.com/Expensify/Expensify.cash/blob/8b979de2cf00deafbba85c5c139f6d78cd1fd3f2/src/pages/iou/IOUModal.js#L294-L297

But that's just the basics. Since we don't hardcode stings I'll also need to add "close" to translations in a similar way it was done in this [PR](https://github.com/Expensify/Expensify.cash/pull/3285/files).

**Comment by stitesExpensify** [comment]
Looks good to me!

**Comment by michaelhaxhiu** [comment]
Looks good to me too! 

@dklymenk going to hire you in Upwork and assign this GH to you. BTW - you have a **great** avatar pic btw, love south park.

**Comment by dklymenk** [comment]
Submitted a [PR](https://github.com/Expensify/Expensify.cash/pull/3351)

**Comment by michaelhaxhiu** [comment]
Set a reminder for my self to pay @dklymenk on June 10 (to allow 7 days for any potential regression discoveries/fixes).

**Comment by iwiznia** [comment]
Found a regression [here](https://github.com/Expensify/Expensify.cash/pull/3351#issuecomment-858823796), please hold on payment

**Comment by michaelhaxhiu** [comment]
Awaiting your thumbs up to pay the contributor @iwiznia

**Comment by iwiznia** [comment]
This is the new PR https://github.com/Expensify/Expensify.cash/pull/3525 and it was already merged.
